impls/python/step2_eval.py

```diff
@@ -4,53 +4,41 @@
 import reader, printer
 
 # read
-def READ(str):
-    return reader.read_str(str)
+READ = reader.read_str
 
 # eval
-def eval_ast(ast, env):
+def EVAL(ast, env):
```

Comment:
Same thing: the guide explains you need a function called `eval_str` which does X,Y,Z, and another function called `EVAL` which does A,B,C . This modification steps away from this guide.

Reply:
In the guide, `eval-ast` serves two very different purposes:
* to shorten the main conditional in `eval`.
* as a helper for  the apply phase and the `do` special form.

Mxing the two does not improve the learning experience. As a beginner discovering step2, I have had the impression that `eval-ast` describes the evaluation of lists as it does for symbols, vectors and maps.
I have suggested to remove `eval-ast` from the guide, but kanaka prefers this structure.
Note however that implementations are allowed to diverge here.  For example, several languages with pattern matching already do, as well as some exotic languages defining `eval` as an overridden object method.
In this merge request, the change is motivated because `do` and the apply phase can manipulate a python generator.. The current structure is the only reason why they construct an intermediate MAL list in memory.